### ISSUE TITLE
Optional concatenation in attribute blocks

### ISSUE THREAD

**Issue by TheNeikos** [lambda-fairy/maud#347]
I'm using tailwindcss, it uses a lot of classes that are annoying to write in rust/maud: `mt-8`, `space-y-6` etc... 

So, I usually resort to write `div class="mt-8 space-y-6"`, and sometimes even

```
div class={
  "mt-8 etc..."
}
```

Now, currently maud supports having concatenating multiple strings like this:

```
div class={
  "mt-8 "
  "space-y-6"
}
```

(notice the extra space in the first, since no additional space is added here sadly)

Now, I would like to be able to have an optional class in this situation. I did not find a suitable feature in this situation, so I would suggest something akin to this:

```
let optional_error_classes: Option<String>;

// ...
div class={
  "mt-8 "
  [optional_error_classes]
}
```

With `None` simply being handled as `""` and the content otherwise.

What do you think?

**Comment by lambda-fairy** [comment]
Hi @TheNeikos!

Sorry about the late response 😅 

Do you know about the class shorthand syntax? We can get something similar with the following:

```rust
let has_space: bool;

// ...
div ."mt-8" ."space-y-6"[has_space]
```

https://maud.lambda.xyz/elements-attributes.html#classes-and-ids-foo-bar
https://maud.lambda.xyz/splices-toggles.html#toggles-foo

If you need to support `Option<String>` specifically, then we could add a `.[some_option]` syntax, similar to the existing `attr=[some_option]` syntax for optional attributes. But I want to check if this is needed first.

**Comment by lambda-fairy** [comment]
@bgrundmann does `.is-primary[state == State::Primary].is-danger[state == State::Danger]` work for you?

**Comment by TheNeikos** [comment]
> Do you know about the class shorthand syntax? We can get something similar with the following:
> 
> ```rust
> let has_space: bool;
> 
> // ...
> div ."mt-8" ."space-y-6"[has_space]
> ```
Yes that is true, but I find writing `.""` very cumbersome for the 10+ classes tailwind sometimes requires.

**Comment by bgrundmann** [comment]
> @bgrundmann does `.is-primary[state == State::Primary].is-danger[state == State::Danger]` work for you?

Sorry for the late reply my GitHub email notifications got lost in my inbox. Yeah that works.  That said the '.[aoptionalclassname]' would I think in the end read better.  But I can understand not wanting to make the language too complicated.

**Comment by lambda-fairy** [comment]
> I would love to have `.[aoptionalclassname]`

Oh that sounds good to me. I don't think it makes the language more complicated, since it's "filling in a hole" (so to speak).

**Comment by mattfbacon** [comment]
Can't you use `@if let Some`?

**Comment by hirschenberger** [comment]
How about:

```rust
#[derive(Clone, Default, Debug, PartialEq, strum_macros::Display)]
pub enum Color {
    #[default]
    #[strum(serialize = "")]
    Default,
    #[strum(serialize = "is-success")]
    Success,
    #[strum(serialize = "is-warning")]
    Warning,
    #[strum(serialize = "is-danger")]
    Danger,
    #[strum(serialize = "is-info")]
    Info,
}

#[derive(Clone, Default, Debug, PartialEq, strum_macros::Display)]
pub enum Size {
    #[strum(serialize = "is-small")]
    Small,
    #[default]
    #[strum(serialize = "")]
    Default,
    #[strum(serialize = "is-medium")]
    Medium,
    #[strum(serialize = "is-large")]
    Large,
}
```

And using it like:

```rust
let state = Color::Danger;
let size = Size::Large;

html! {
  .message (size) (state) {
    .message-body {
      "Hamburgefontsiv" 
    }
  }
}
```

**Comment by avdb13** [comment]
Updates? Attempting to use Maud with Tailwind too.

**Comment by Wragdan** [comment]
I added the tailwind build to my build step:
```rust
// build.rs
fn main() {
    let tailwind_cmd = "npx tailwindcss -i index.css -o assets/index.css";

    let path = fs::canonicalize(".");

    let path = match path {
        Ok(p) => p,
        Err(e) => panic!("{}", e),
    };

    std::process::Command::new("zsh")
        .current_dir(path)
        .arg("-c")
        .arg(tailwind_cmd)
        .spawn()
        .expect("failed running tailwind");
}
```

Not the best code, but I'm still in dev, so not production code for now.
If you run `cargo watch -x run` it compiles on each file change.

**Comment by re-oh** [comment]
@avdb13 solved something simillar to this just today, although its a bit of a different case i think that it could be used in this context check issue: #446 and pull: #447
